unlock-app/src/components/creator/CreatorLog.tsx

```diff
@@ -0,0 +1,72 @@
+import React from 'react'
+import styled from 'styled-components'
+import * as UnlockTypes from '../../unlock'
+
+interface Props {
+  transactionFeed: UnlockTypes.Transaction[]
+  transactionMetadata: UnlockTypes.TransactionMetadataMap
+}
+
+const CreatorLog = (props: Props) => {
+  const { transactionFeed, transactionMetadata } = props
+  return (
+    <Grid>
+      <HeaderItem>Block Number</HeaderItem>
+      <HeaderItem>Lock Name/Address</HeaderItem>
+      <HeaderItem>Type</HeaderItem>
+      {transactionFeed.map(tx => {
+        const { href, readableName } = transactionMetadata[tx.hash]
+        return (
+          <React.Fragment key={tx.hash}>
```

Comment:
All of this could be a `<Transaction>` component which would display the readable name without having to pass it at state.

Reply:
Can you elaborate a bit on this?